src/cookies.js

```diff
@@ -253,8 +253,9 @@ function checkOriginForSettingCookie(win, options, name) {
   );
   const current = parseUrlDeprecated(win.location.href).hostname.toLowerCase();
   const proxy = parseUrlDeprecated(urls.cdn).hostname.toLowerCase();
+  const metaTag = win.document.querySelector("meta[name='runtime-host']");
   userAssert(
-    !(current == proxy || endsWith(current, '.' + proxy)),
+    !(current == proxy || endsWith(current, '.' + proxy) || metaTag != null),
```

Comment:
sorry for the delays, since its holiday season a few people are out and I need some additional pairs of eyes on this since i know the change is small but since its about cookies i'd like to tread carefully.

so in this case, basically the publisher is doing an opt-in to allow for cookie writes? would this also allow writing to the AMP cache domain? (publisher-domain.ampproject.org/*, google.com/amp/?)

Reply:
That's a great question @erwinmombay. As it stands today, I believe any page that contains `<meta name="runtime-host">` fails validation: #27546 and doc [amp-framework-hosting](https://github.com/ampproject/amphtml/blob/main/docs/spec/amp-framework-hosting.md). @sebastianbenz , do you know if this is still the case? In this sense, there probably isn't much worry about cookie writes to the AMP cache domain since pages won't end up there anyways. With that said, it's probably best not to rely on this always being the case since #27546 is an open request (though, I suspect these meta tags would be stripped from the page before caching).

I believe the `userAssert(!isProxyOrigin(win.location.href), ...)` check just above this one is going to protect against setting cookies on publisher-domain.cdn.ampproject.org and cdn.ampproject.org just fine. So, even for the cases of
`<meta name="runtime-host" value="">` and `<meta name="runtime-host" value="https://cdn.ampproject.org">`, which both result in `urls.cdn = https://cdn.ampproject.org`, cdn.ampproject.org should be safe.

I'm a little unclear on why this "in depth" `!(current == proxy || endsWith(current, '.' + proxy))` check is performed. It dates back pretty far and there wasn't discussion about it: https://github.com/ampproject/amphtml/pull/8690 . It seems to be saying "don't trust `cdnProxyRegex`, repeat the check against `cdn`". I'm inclined to suggest it be dropped. I _think_ the spirit of the check is still "is this a proxy origin?", which the previous check covers. Would love confirmation from others.

Finally, if this "in depth" check is worthwhile, then we should probably think about allowed URLs in the meta tag's `value`, instead of just checking for existence of the tag itself.

Reply:
I agree @mdmower. It definitely is dated.  If this "in depth" check is still desired then we definitely should assess what URLS should be allowed in the meta tag's value as stated by mdmower. If it is not desired and it is decided that we part ways with this check, I would love to discuss what would replace it and take on the role of implementing it. Eager to hear back so we can get this PR rolling! and Thank you @mdmower

Reply:
i looked into the validator and it seems it allows `<meta name="runtime-host" content="https://example.com">` but i can't seem to find where that is allowed. digging into the validator some more and asking for additional pair of eyes on this

Reply:
@erwinmombay - I'd love to be included in the conversation if you and others plan to hash this out over slack or video. From a practical standpoint, I think the ultimate goal of this PR makes sense: whereas AMP Project hosted pages need to discern whether they are being viewed from a viewer (like a search engine's AMP viewer) or are being accessed directly from the CDN to determine whether analytics cookies should be set, self-hosted AMP frameworks typically live on the same domain as the AMP pages (it's an [advertised benefit of self-hosting](https://github.com/ampproject/amphtml/blob/main/docs/spec/amp-framework-hosting.md)). The original check `!(current == proxy || endsWith(current, '.' + proxy))` from 2017 didn't have self-hosting in mind, and I'm still of the opinion that the check is unnecessary.

Reply:
yes. lets have a call to make quicken the process a little bit. let me get a hold of some people who might have some input on this. what are good days for y'all to have the call?

we're also waiting for us to sync our external validator with the latest code

Reply:
Thanks! For me, if M-F then these Pacific times are available:

- 7:30-11am
- 12pm - 1:30pm
- 2:15pm - evening

Weekends are possible if that's what works best.

Reply:
Awesome! For me, if during the work week then these time blocks work for me (central time):

- 7:30am - 12:00pm
- 2:15pm - evening

Weekends are possible as well. I am looking forward to the call!

Reply:
thanks all. I'll wait on @sebastianbenz feedback first to see if this is necessary. I am pretty much OK with merging this as long as the external validator has been updated with the `meta` rules. (I'll confirm that today)

Reply:
Awesome! Looking forward to hearing back. If this does get merged in I would be more than willing to update some documentation. Let me know!

Reply:
What I don't understand : how does this enable cookies? 

Given: 

```
current = example.com
proxy = example.com
metaTag = true
``` 

Wouldn't in this case the assertion still throw an error? 

```
 userAssert(
    !(current == proxy || endsWith(current, '.' + proxy) || metaTag != null),
    'Should never attempt to set cookie on proxy origin. (in depth check): ' +
      name
  );
```

I'm sure I'm misunderstanding something, would be great if someone could explain this to me.

Reply:
I spent most of my time advocating for removal of the assert altogether and didn't scrutinize the logic. Thank you for looking closely @sebastianbenz, I don't think you're missing anything (unless I am, too).

@JakeSquier - it sure looks like this code change would be ineffective, could you test again? @erwinmombay previously suggested adding a unit test; I think that is needed.

Also, it looks like you need to let vscode autoformat your code, or use `amp lint --local_changes --fix`. Documentation: https://github.com/ampproject/amphtml/blob/main/docs/getting-started-e2e.md#code-quality-and-style

Reply:
@mdmower - No problem it will be taken care of. I will add a unit test and also take care of the ineffective code. This PR will be updated sometime today; Thank you for pointing these things out!